controllers/postgres_controller.go

```diff
@@ -1062,14 +1063,28 @@ func (r *PostgresReconciler) checkAndUpdatePatroniReplicationConfig(log logr.Log
 			return requeueAfterReconcile, nil
 		}
 		if instance.Spec.PostgresConnection.SynchronousReplication {
-			if resp.SynchronousNodesAdditional == nil || *resp.SynchronousNodesAdditional != instance.Spec.PostgresConnection.ConnectedPostgresID {
+			// fetch the sync standby to determine the correct application_name of the instance
+			log.V(debugLogLevel).Info("fetching the referenced sync standby")
+			var synchronousStandbyApplicationName *string
+			s := &pg.Postgres{}
+			ns := types.NamespacedName{
+				Name:      instance.Spec.PostgresConnection.ConnectedPostgresID,
+				Namespace: instance.Namespace,
+			}
+			if err := r.CtrlClient.Get(ctx, ns, s); err != nil {
+				r.recorder.Eventf(s, "Warning", "Error", "failed to get referenced sync standby: %v", err)
+				synchronousStandbyApplicationName = nil
+			} else {
+				synchronousStandbyApplicationName = pointer.String(s.ToPeripheralResourceName())
+			}
+			if resp.SynchronousNodesAdditional == nil || *resp.SynchronousNodesAdditional != *synchronousStandbyApplicationName {
```

Comment:
a few lines above you have

~~~
if err := r.CtrlClient.Get(ctx, ns, s); err != nil {
  r.recorder.Eventf(s, "Warning", "Error", "failed to get referenced sync standby: %v", err)
  synchronousStandbyApplicationName = nil
~~~
in this case you set the variable `synchronousStandbyApplicationName` to `nil` and continue. but in this following line you check against the value `*synchronousStandbyApplicationName` but this will crash if the upper code sets the value to `nil`